### ISSUE TITLE
[Proposal] Codelists requiring descriptions

### ISSUE THREAD

**Issue by odscjen** [GFDRR/rdl-standard#112]
## What is the context or reason for the change?
A number of existing codelists do not currently have descriptions and some of the new ones to be created will also need descriptions.

## What is your proposed change?

Add descriptions to the following codelists:

- [x] IMT.csv - descriptions proposed in new branch see https://github.com/GFDRR/rdl-standard/pull/114
- [x] Hazard_type.csv - [see #130](https://github.com/GFDRR/rdl-standard/pull/130)
- [x] process_type.csv -- see https://github.com/GFDRR/rdl-standard/issues/112#issuecomment-1622524165
- [x] seasonality.csv -- see https://github.com/GFDRR/rdl-standard/issues/112#issuecomment-1623867795
- [x] cost_type.csv (currently in PR #101)
- [x] frequency_type.csv (still to be created) - (Probabilistic, Deterministic, Empirical) same as analysis_type.csv
- [x] calculation_method.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/73)) same as base_data_type.csv
- [x] spatial_scale.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/77)) proposed in https://github.com/GFDRR/rdl-standard/issues/77#issuecomment-1613045581
- [x] analysis_type,csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/72)) proposed in https://github.com/GFDRR/rdl-standard/issues/72#issuecomment-1613352187
- [x] function_approach.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/78)) - (Empirical survey, Analytical, Judgement, Hybrid) proposed in https://github.com/GFDRR/rdl-standard/issues/78#issuecomment-1613223352
- [x] base_data_type.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/75)) - (Simulated, Inferred, Observed) proposed in https://github.com/GFDRR/rdl-standard/issues/75#issuecomment-1613213583
- [x] impact_type.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/75))  - (direct, indirect, total) proposed in https://github.com/GFDRR/rdl-standard/issues/75#issuecomment-1613079878
- [x] impact_metric.csv (([still to be created](https://github.com/GFDRR/rdl-standard/issues/125)) proposal at https://github.com/GFDRR/rdl-standard/issues/125#issuecomment-1622449558 for TWO TABLES: one for vulnerability and one for loss
- [x] engineering_demand_parameter ([still to be created](https://github.com/GFDRR/rdl-standard/issues/78)) proposed at https://github.com/GFDRR/rdl-standard/issues/78#issuecomment-1613395210
- [x] damage_scale_names ([still to be created](https://github.com/GFDRR/rdl-standard/issues/78)) proposed in https://github.com/GFDRR/rdl-standard/issues/78#issuecomment-1613474557
- [x] relationship_type.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/78)) - (mathematical function (parametric), mathematical function (bespoke), discrete values) proposed in https://github.com/GFDRR/rdl-standard/issues/78#issuecomment-1613438953
- [ ] damage_states_name.csv ([still to be created](https://github.com/GFDRR/rdl-standard/issues/78)) -- see https://github.com/GFDRR/rdl-standard/issues/112#issuecomment-1622458838

Others may be added to this list as we work through the other issues.

**Comment by odscjen** [comment]
@stufraser1 I think this is the codelists that need descriptions that you're input would be really good for.

Where there's a link to either an issue in the above list if you could just add a table in a comment in the issue with the descriptions that would be great.

For the one that's in a current PR you can just update the file and push to that PR.

For the ones that already exist if you just create a new branch and update them there.

**Comment by odscjen** [comment]
Just added a couple more to the list :)

**Comment by stufraser1** [comment]
frequency_type.csv has the same codes as analysis_type. Use same definitions in different codelist, same codelist?

**Comment by stufraser1** [comment]
calculation_method has the same codes as base_data_type. Use same definitions in different codelist, same codelist?

**Comment by stufraser1** [comment]
I'm not sure we should create a codelist for damage_scale_names.
It is more of a user entered field, depending on which reference they use. Certainly would be open codelist but isn't really a definition of terms.

**Comment by stufraser1** [comment]
> Happy to leave damage_scale_names as a free text field.

On reflection, there was a set of enum fields defined by the vulnerability experts who developed the schema, I opted to recreate that table as a codelist to use.

**Comment by odscjen** [comment]
Great, I'll mark it as an open codelist

**Comment by odscjen** [comment]
2 more added

**Comment by odscrachel** [comment]
added seasonality.csv

**Comment by stufraser1** [comment]
re damage_states_name.csv I think this should remain a string without a codelist. The names used for each damage state do vary by damage scale used and can be written in from that source information. It would be challenging to write a codelist and define each term, because each scale may use a different definition for the damage state (minor damage may mean something different based on how it is assessed, the hazard in question and the number of other states in the scale.

**Comment by stufraser1** [comment]
> added seasonality.csv

I don't think a codelist is needed - I'm not sure what function it would serve for seasonality, which refers to a table of events that defines, likely numerically, the time in a year that the event occurs - in order to simulate the peak times of year in which events can occur.

**Comment by stufraser1** [comment]
process_type.csv
For some reason I was not able to edit the csv file directly, so here is the suggestion:

Code,Title,Description
FCF,Coastal Flood,"Coastal flooding is most frequently the result of storm surges and high winds coinciding with high tides. The surge itself is the result of the raising of sea levels due to low atmospheric pressure. In particular configurations, such as major estuaries or confined sea areas, the piling up of water is amplified by a combination of the shallowing of the seabed and retarding of return flow (WMO, 2011)."
FSS,Storm Surge,"A storm surge reflects the difference between the actual water level under the influence of a meteorological disturbance (storm tide) and the level which would have occurred in the absence of the meteorological disturbance (i.e., astronomical tide) (WMO, 2008, 2011, 2017)."
TOR,Tornado,"A tornado is a rotating column of air, extending from the base of a cumuliform cloud, and often visible as a condensation funnel in contact with the ground, and/or attendant circulating dust or debris cloud at the ground (WMO, 2017)."
DTA,Agricultural Drought,"Agricultural drought: Occurs when agricultural production becomes affected. It focuses on precipitation shortages, differences between actual evapotranspiration, soil water deficits, reduced groundwater and so on."
DTH,Hydrological Drought,"Hydrological drought occurs when low water supply becomes evident and is associated with the effects of periods of precipitation shortfalls on surface or subsurface water supply."
DTM,Meteorological Drought,"Meteorological drought occurs when dry weather patterns dominate an area. It is defined usually on the basis of the degree of dryness and the duration of the dry period."
DTS,Socio-economic Drought,"Socioeconomic drought: Relates the supply and demand of some economic goods with elements of meteorological, hydrological, and agricultural drought. It also occurs when the demand for an economic good exceeds supply as a result of a weather-related shortfall in water supply."
Q1R,Primary Rupture,"Earthquake surface ruptures and fissures are localised ground displacements that develop during and immediately after an earthquake. Surface ruptures represent the upward continuation of fault slip at depth, while fissures are smaller displacements, or more distributed deformation in and around the rupture area."
Q2R,Secondary Rupture,
QGM,Ground Motion,Earthquake ground shaking is the movement of the Earth’s surface produced by seismic waves that are generated when an earthquake occurs.
QLI,Liquefaction,"Soil liquefaction occurs when soil is transformed from a solid to a liquid state as a result of increased pore pressure and reduced effective stress. It is typically caused by rapid loading of the soil during earthquake shaking (AGI, 2017)."
ECD,Extreme cold,"A cold wave is a period of marked and unusual cold weather characterised by a sharp and significant drop in air temperatures near the surface (maximum, minimum and daily average) over a large area and persisting below certain thresholds for at least two consecutive days during the cold season (WMO, 2020)."
EHT,Extreme heat,"A heatwave is a marked unusual period of hot weather over a region persisting for at least two consecutive days during the hot period of the year based on local climatological conditions, with thermal conditions recorded above given thresholds (WMO, 2020)."
FFF,Fluvial Flood,"A fluvial flood is a rise, usually brief, in the water level of a stream or water body to a peak from which the water level recedes at a slower rate (WMO, 2012)."
FPF,Pluvial Flood,"Pluvial flood or surface water flooding is caused by rain remaining and ponding on the ground surface during and after rainfall events, and can also include urban flooding, which results form the capacity of urban drainage systems being exceeded."
FGF,Groundwater Flood,"A groundwater flood is the emergence of groundwater at the ground surface away from perennial river channels or the rising of groundwater into man-made ground, under conditions where the ‘normal’ ranges of groundwater level and groundwater flow are exceeded (BGS, 2010)."
LAV,Snow Avalanche,"An avalanche is a mass of snow and ice falling suddenly down a mountain slope and often taking with it earth, rocks and rubble of every description (WMO, 1992)."
LSL,Landslide (general),"Landslide is the downslope movement of soil, rock and organic materials under the effects of gravity, often triggered by seismic activity or precipitation. This general code can be used to include multiple specific types of landslide activity"
LSR,Rockslide,"A rock slide is a movement of a mass of soil or rock on an individualized failure surface (Dennis and Didier, 2019)."
LSM,Mudflow,"A mud flow is a flow of water so heavily charged with sediment and debris that the flowing mass is thick and viscous (WMO and UNESCO, 2012)."
LSF,Rockfall,"Rockfall is a fragment of rock (a block) detached by sliding, toppling, or falling, that falls along a vertical or sub-vertical cliff."
TSI,Tsunami,"A tsunami is a series of travelling waves of extremely long length and period, usually generated by disturbances associated with earthquakes occurring below or near the ocean floor, but can also be caused by landslides (subaerial and submarine), volcanic activity and meteorological conditions."
VAF,Ashfall,"Volcanic ash refers to the finest particles of tephra (less than 2 mm diameter) - tephra is a collective term for fragmented magma and old (i.e., preexisting) rocks ejected into the atmosphere from volcanic vents during an explosive eruption."
VBL,Ballistics,"Ballistics comprise fragments of magma and old (i.e., pre-existing) rocks ejected during an explosive eruption."
VLH,Lahar,"Lahars are discrete, rapid, gravity-driven, water-saturated flows containing water and solid particles of volcanic rock, sediment, ice, wood, and other debris that originate at volcanoes."
VLV,Lava,"A lava flow forms during an eruption or eruptive episode and travels away from the source vent as fluid, relatively low-viscosity molten rock."
VPF,Pyroclastic Flow,"Also known as pyroclastic density currents these are hot, fast-moving mixtures of volcanic particles and gas that originate from the gravitational collapse of explosive eruption columns, lava domes or lava-flow fronts, and from explosive lateral blasts."
WFI,Wildfire,"Wildfires are any unplanned or uncontrolled fire affecting natural, cultural, industrial and residential landscapes (UNDRR adapted from FAO, 2010)."
ETC,Extratropical cyclone,"An extra-tropical cyclone is a low-pressure system which develops in latitudes outside the tropics (WMO, 1992)."
TCY,Tropical cyclone,"A tropical cyclone is a warm-core, non-frontal synoptic-scale cyclone, originating over tropical or subtropical waters, with organised deep convection and closed surface wind circulation about a well-defined centre (WMO, 2017)."
MHC,Multi-Hazard combined,

__________

Intended commit comments:
Updated descriptions for process_type.
Largely using UNDRR Hazard terminology as the basis (included citations come from that document) and we should reflect this as the source in the overall documentation.

We need to confirm the difference between primary and secondary rupture.
We need to confirm whether we need code MHC, given we discussed putting multiple process types together in a string.
I added 3 specific landslide types.

**Comment by stufraser1** [comment]
I think I've completed all the necessary tasks (remaining unchecked codelists not required, see above comments. Not all created as .csv files so please see the comments where I've proposed structure to create those csvs.

**Comment by odscjen** [comment]
Great, thanks. As per the comments on the hazard_type PR can you update the process_type codes to be full words rather than abbreviations?

**Comment by stufraser1** [comment]
comment above ready for you @odscjen

**Comment by stufraser1** [comment]
Discussed https://github.com/GFDRR/rdl-standard/issues/112#issuecomment-1622462306 offline with @odscrachel.

Agreed to provide codelist as follows:

Code,Title,Description
uniform,Uniform,"Denotes that events are simulated without a defined seasonal distribution - i.e. an event has an equal chance of occurring at any time of year, which is relevant for geohazards such as earthquake or tsunami."
user_defined,User-defined,"Denotes that a seasonal distribution is described in an associated resource file. A model uses this distribution to ensure simulated events are more likely to occur in the seasons which match reality - for example the peak months for cyclone, drought and flood occurrence."

**Comment by stufraser1** [comment]
Noted three csv codelist files [here](https://github.com/GFDRR/rdl-standard/tree/dev/specs/code-lists/CSV) have no description - do we need to update these now?

**Comment by odscjen** [comment]
occupancy.csv is no longer required as we agreed to remove occupancy from the schema.
frequency_type.csv is also no longer required due to the rearranging of the occurrence fields.
val_type.csv has been replaced with [codelists/closed/cost_type.csv](https://github.com/GFDRR/rdl-standard/blob/dev/codelists/closed/cost_type.csv)

So we can delete all 3 once all the changes have been made and we've confirmed that everyone's happy with them.
